psq/datastore_storage.py

```diff
@@ -40,16 +42,31 @@ def __init__(self, datastore):
         _check_for_thread_safety(datastore)
         self.datastore = datastore
 
+    @retry(
```

Comment:
Why not make one global retry decorator and re-use it, that way we don't have to repeat ourselves:

```python
_RETRY = retry(stop_max_attempt_number=5, ...)

...

@_RETRY
def get_task(self, task_id):
    ...
```